rslib/src/scheduler/reviews.rs

```diff
@@ -100,12 +109,33 @@ impl Collection {
         let today = self.timing_today()?.days_elapsed;
         let mut rng = rand::thread_rng();
         let distribution = Uniform::from(spec.min..=spec.max);
+        let mut decks_initial_ease: HashMap<DeckId, f32> = HashMap::new();
         self.transact(Op::SetDueDate, |col| {
             col.storage.set_search_table_to_card_ids(cids, false)?;
             for mut card in col.storage.all_searched_cards()? {
+                let deck_id = card.original_deck_id.or(card.deck_id);
+                let ease_factor = match decks_initial_ease.get(&deck_id) {
+                    Some(ease) => *ease,
+                    None => {
+                        let config_id = col
+                            .get_deck(deck_id)?
+                            .ok_or(AnkiError::NotFound)? // TODO: use the right error
```

Comment:
I wasn't quite sure what errors would be reasonable to return here -- should I just go with `NotFound` for all of them, or should I return some kind of string-based error in all of these cases?

Reply:
For issues that the DB check will clean up, a NotFound error is fine. IIRC, it handles missing decks and incorrect original_deck_ids, but does not handle missing deck configurations. We should probably address that in the future, but for now I'd recommend you just enable the fallback option on get_deck_config()

Reply:
Will do, thanks!